src/backend/distributed/utils/citus_stat_tenants.c

```diff
@@ -221,41 +233,86 @@ void
 AttributeTask(char *tenantId, int colocationId, CmdType commandType)
 {
 	if (StatTenantsTrack == STAT_TENANTS_TRACK_NONE ||
-		tenantId == NULL || colocationId == INVALID_COLOCATION_ID)
+		colocationId == INVALID_COLOCATION_ID)
 	{
 		return;
 	}
 
+	/*
+	 * if tenantId is NULL, it must be a schema-based tenant and
+	 * we try to get the tenantId from the colocationId to lookup schema name and use it as a tenantId
+	 */
+	if (tenantId == NULL)
+	{
+		if (!IsTenantSchemaColocationGroup(colocationId))
+		{
+			return;
+		}
+
+	}
```

Comment:
I don't think this check is needed anymore. We only annotate the query if with tenantId=NULL if IsTenantSchemaColocationGroup returned true. So no need to re-check it here again.

Reply:
You're right, except for locally executed queries (ex: queries to reference tables) that directly follow the AttributeTask(..) flow without being annotated. We can move this check to [local_executor](https://github.com/citusdata/citus/blob/gokhangulbiz/schema-based-tenant-monitoring/src/backend/distributed/executor/local_executor.c#L692). Let me know your thoughts.